doc/develop/west/without-west.rst

```diff
@@ -69,6 +69,10 @@ If you don't have west installed and your application *does* need one
 of these repositories, you must set :makevar:`ZEPHYR_MODULES`
 yourself as shown above.
 
+The implementation CMake uses to set :makevar:`ZEPHYR_MODULES` can be
+found in the Zephyr source code in  :file:`cmake/modules/zephyr_module.cmake`.
+It is primarily a regex which parses the result of ``west list``
```

Comment:
I think you are basing this statement off of the following comment in zephyr_module.cmake:

```
# If west is available, it uses `west list` to obtain a list of projects to
# search for zephyr/module.yml
```

This is long-stale information. The CMake module gets its job done by invoking a python script which uses the west API to programmatically decide which west projects are also zephyr modules; no regular expressions on west list output involved.

Reply:
> a high-level description of the logic implemented in this script would be awesome.

It's just a loop over all the west.manifest.Project instances returned by the west API, looking for repositories which contain the module.yml files documented in https://docs.zephyrproject.org/latest/develop/modules.html#module-yaml-file-description

Reply:
I mean, I say "just" -- it obviously isn't "just" that in reality, but it is in fact "just" that from a high level.

Reply:
> If you can make sense of it,...

I never looked at this script; I should have clarified this.

> looking for repositories which contain the module.yml files documented in

Thanks, it's probably worth having this link and sentence here.

Reply:
It's quite difficult, and I need to figure out how to word it. During the CMake `./configure` phase, CMake will create `build/zephyr_modules.txt`, the way that it does this is unclear to me, and it seems like I might be wrong about the regex scanning of `west list`.